lib/kubernetes-deploy/kubernetes_resource/job.rb

```diff
@@ -13,6 +13,9 @@ def deploy_succeeded?
     def deploy_failed?
       return false unless deploy_started?
       return false unless @instance_data.dig("spec", "backoffLimit").present?
+      (@instance_data.dig("status", "conditions") || []).each do |condition|
+        return true if condition["type"] == 'Failed' && condition['status'] == "True"
+      end
```

Comment:
Nit: can you extract this into a method?

This condition also has richer information about the reason for the failure in its reason/message fields. Let's adjust `failure_message` to include that information if we have it.

Can you also add tests? Maybe unit tests that show `deploy_failed?` is true with either of the statuses I discovered are possible on the issue?